src/devtools/client/debugger/src/components/Editor/EmptyLines.tsx

```diff
@@ -1,109 +1,92 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+import { useRef, useMemo, useEffect } from "react";
 
-import { connect, ConnectedProps } from "react-redux";
+import { useAppSelector } from "ui/setup/hooks";
 
-import { Component } from "react";
 import { fromEditorLine } from "../../utils/editor";
 import { getBreakableLinesForSelectedSource } from "ui/reducers/possibleBreakpoints";
-import { getBoundsForLineNumber } from "ui/reducers/hitCounts";
-import type { UIState } from "ui/state";
+
+import { getSelectedSourceId } from "ui/reducers/sources";
 
 import { calculateHitCountChunksForVisibleLines } from "devtools/client/debugger/src/utils/editor/lineHitCounts";
-import { editorItemActions } from "./menus/editor";
 
 import type { SourceEditor } from "../../utils/editor/source-editor";
 
-const mapStateToProps = (state: UIState) => {
-  const breakableLines = getBreakableLinesForSelectedSource(state);
-
-  const { lower, upper } = getBoundsForLineNumber(state.app.hoveredLineNumberLocation?.line || 0);
-
-  return {
-    breakableLines,
-    lower,
-    upper,
-  };
-};
-
-const connector = connect(mapStateToProps);
-
-type PropsFromRedux = ConnectedProps<typeof connector>;
-
 interface ELProps {
   editor: SourceEditor;
 }
 
-type FinalELProps = PropsFromRedux & ELProps;
-
-class EmptyLines extends Component<FinalELProps> {
-  _animationFrameId: number | null = null;
+export default function EmptyLines({ editor }: ELProps) {
+  const breakableLines = useAppSelector(getBreakableLinesForSelectedSource);
+  const animationFrameId = useRef<number | null>(null);
```

Comment:
This ref isn't necessary, is it? You only reference the values it holds inside of the effect, so you should just be able to use a local variable.